hera_pspec/pspecdata.py

```diff
@@ -642,27 +649,36 @@ def C_model(self, key, model='empirical', time_index=None, known_cov=None):
                 # add model to key
                 Ckey = ((dset, dset), (bl,bl), ) + (model, time_index, False, True,)
 
+        # Update self._C with known_cov
+        if known_cov is not None:
+            spw = slice(self.get_spw(include_extension=include_extension)[0], self.get_spw(include_extension=include_extension)[1])
```

Comment:
just FYI, this can be done more elegantly as `spw = slice(*self.get_spw(include_extension=include_extension))`. I'm not insistent that you change it, but just wanted to let you know :)

Reply:
Thanks! I will modify it.